src/exporters/s5.py

```diff
@@ -0,0 +1,492 @@
+
+        variable: str
+            the variable that you want to download.
+
+        pressure_level: bool
+            Do you want data at different atmospheric heights?
+            True = yes want data at different `pressure-levels`
+            False = no want only `single-level` data
+
+        granularity: str, {'monthly', 'hourly'}, default: 'monthly'
+            The granularity of data being pulled.
+            'hourly'/'pressure_level' data has a forecast for every 12hrs (12, 24, 36 ...)
+            'hourly'/'single_level' data has a forecast for every 6hrs (6, 12, 18, 24 ...)
+            'monthly' data has a forecast for every month {1 ... 6}
+
+        product_type : str
+            The product type is only valid for monthly datasets. It corresponds
+            to the post-processing of the monthly forecasts
+            Defaults to 'monthly_mean' if no product_type is given
+                if pressure_level == True:
+                 {'ensemble_mean', 'hindcast_climate_mean', 'monthly_mean'}
+                else
+                 {'ensemble_mean', 'hindcast_climate_mean', 'monthly_mean',
+                  'monthly_standard_deviation', 'monthly_maximum', 'monthly_minimum'}
+
+        Notes:
+        -----
+        - because the `dataset` argument is not intuitive, the s5 downloader
+        makes decisions for the user based on their preferences in the args
+        `pressure_level` and `granularity`.
+        - these are constant for one download
+        """
+        super().__init__(data_folder)
+
+        # initialise attributes for this export
+        self.pressure_level = pressure_level
+        self.granularity = granularity
+
+        if dataset is None:
+            self.dataset = self.get_dataset(self.granularity, self.pressure_level)
+        else:
+            self.dataset = dataset
+
+        # get the reference dictionary that corresponds to that dataset
+        self.dataset_reference = dataset_reference[self.dataset]
+
+        # get the product type if it exists
+        self.product_type = self.get_product_type(product_type)
+
+        # check the product_type is valid for this dataset
+        if self.dataset_reference["product_type"] is not None:
+            assert (
+                self.product_type in self.dataset_reference["product_type"]
+            ), f"\
+                {self.product_type} is not a valid variable for the {self.dataset} dataset.\
+                Try one of: {self.dataset_reference['product_type']}"
+
+    def export(
+        self,
+        variable: str,
+        min_year: int = 2017,
+        max_year: int = 2018,
+        min_month: int = 1,
+        max_month: int = 12,
+        max_leadtime: Optional[int] = None,
+        pressure_levels: Optional[List[int]] = None,
+        selection_request: Optional[Dict] = None,
+        N_parallel_requests: int = 3,
+        show_api_request: bool = True,
+    ):
+        """
+        Arguments
+        --------
```

Comment:
Nit: This docstring is missing some of the parameters

Reply:
thanks